src/libraries/Microsoft.PowerFx.Core/Public/Values/DateTimeValue.cs

```diff
@@ -15,6 +15,46 @@ namespace Microsoft.PowerFx.Types
     /// </summary>
     public class DateTimeValue : PrimitiveValue<DateTime>
     {
+        /// <summary>
+        /// Direct use of <see cref="Value"/> is prohibited in favor of <see cref="GetConvertedValue(TimeZoneInfo)"/> method.
+        /// </summary>
+        public new DateTime Value => throw new NotSupportedException(string.Format("Use Method {0} instead, with proper timezone information.", nameof(GetConvertedValue)));
+
+        /// <summary>
+        /// Converts To UTC time if value is not utc and <paramref name="timeZoneInfo"/> is utc.
+        /// Converts from UTC time if  value is utc and <paramref name="timeZoneInfo"/> is non utc.
+        /// else returns the  value/>/>.
+        /// NOTE: if <paramref name="timeZoneInfo"/> is null, Local time zone is used.
+        /// </summary>
+        public DateTime GetConvertedValue(TimeZoneInfo timeZoneInfo)
```

Comment:
Just so i understand, so if someone set `symbolValues.SetTimeZone()` , it isn't really honored here, and they still have to call GetConvertedValue() everytime they read a DateTime value?

Reply:
That is correct, But For now, you can still use Value property directly but we are marking it Obsolete.